packages/cli/src/generators/index.js

```diff
@@ -0,0 +1,127 @@
+const path = require('path');
+
+const prettier = require('gulp-prettier');
+
+const {
+  NODE_VERSION,
+  PACKAGE_VERSION,
+  PLATFORM_PACKAGE
+} = require('../constants');
+
+const Generator = require('yeoman-generator');
+
+const AUTH_TYPE_CHOICES = [
+  {
+    name: 'Basic',
+    value: 'basic'
+  },
+  {
+    name: 'Custom (API Key)',
+    value: 'custom'
+  },
+  {
+    name: 'Digest',
+    value: 'digest'
+  },
+  {
+    name: 'OAuth1',
+    value: 'oauth1'
+  },
+  {
+    name: 'OAuth2',
+    value: 'oauth2'
+  },
+  {
+    name: 'Session',
+    value: 'session'
+  }
+];
+
+class ProjectGenerator extends Generator {
+  initializing() {
+    this.sourceRoot(path.resolve(__dirname, 'templates'));
+    this.destinationRoot(path.resolve(this.options.path));
```

Comment:
These are parts of the Yeomen Generator API: https://yeoman.io/authoring/file-system.html

- `sourceRoot` is where we keep the templates. We set it to `packages/cli/generators/templates`.
- `destinationRoot` is the directory of the new project.